sftp/sftp_test.go

```diff
@@ -187,6 +188,46 @@ func TestDelete(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestRetry(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockFileManager := mock_sftp.NewMockFileManager(ctrl)
+	mockFileManager.EXPECT().Reset().Return(nil)
+
+	callCounter := 0
+	mockFileManager.EXPECT().Upload(gomock.Any(), gomock.Any()).Return(nil).DoAndReturn(func(_, _ interface{}) error {
```

Comment:
can you try this:
mockClient.EXPECT().
    Do(gomock.Any()).
    Return(errors.New("connection lost")). //first
    Return(nil) //second

Reply:
You mean mock mockFileManager? Also, Do() is not supported for mockFileManager

Reply:
chaining on Return() is not supported.

Reply:
ok